examples/hetero/ogb_mag240m_graphsage.py

```diff
@@ -0,0 +1,450 @@
+from torch_geometric.data import Batch
+from torch_geometric.data.lightning import LightningNodeData
+from torch_geometric.loader.neighbor_loader import NeighborLoader
+from torch_geometric.nn import SAGEConv, to_hetero
+from torch_geometric.typing import Adj, EdgeType, NodeType
+
+
+class MAG240M(LightningNodeData):
+    def __init__(self, *args, **kwargs):
+        super(MAG240M, self).__init__(*args, **kwargs)
+
+    @property
+    def num_features(self) -> int:
+        return 768
+
+    @property
+    def num_classes(self) -> int:
+        return 153
+
+    def metadata(self) -> Tuple[List[NodeType], List[EdgeType]]:
+        node_types = ["paper", "author", "institution"]
+        edge_types = [
+            ("author", "affiliated_with", "institution"),
+            ("institution", "rev_affiliated_with", "author"),
+            ("author", "writes", "paper"),
+            ("paper", "rev_writes", "author"),
+            ("paper", "cites", "paper"),
+        ]
+        return node_types, edge_types
+
+
+class HomoGNN(torch.nn.Module):
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        hidden_channels: int,
+        num_layers: int,
+        heads: int = 4,
+        dropout: float = 0.5,
+    ):
+        super().__init__()
+        self.dropout = dropout
+        self.num_layers = num_layers
+
+        self.conv1 = SAGEConv(in_channels, hidden_channels)
+        self.conv2 = SAGEConv(hidden_channels, hidden_channels)
+        self.lin = Linear(hidden_channels, out_channels)
+
+    def forward(self, x: Tensor, edge_index: Adj) -> Tensor:
+        x = x.to(torch.float)
+        x = self.conv1(x, edge_index).relu()
+        x = F.dropout(x, p=self.dropout, training=self.training)
+        x = self.conv2(x, edge_index).relu()
+        x = F.dropout(x, p=self.dropout, training=self.training)
+        return self.lin(x)
+
+
+class HeteroGNN(torch.nn.Module):
+    def __init__(
+        self,
+        metadata: Tuple[List[NodeType], List[EdgeType]],
+        in_channels: int,
+        out_channels: int,
+        hidden_channels: int,
+        num_nodes_dict: Dict[NodeType, int],
+        num_layers: int,
+        heads: int = 4,
+        dropout: float = 0.5,
+        debug: bool = False,
+    ):
+        super().__init__()
+        model = HomoGNN(
+            in_channels,
+            out_channels,
+            hidden_channels,
+            num_layers,
+            heads=heads,
+            dropout=dropout,
+        )
+        self.model = to_hetero(model, metadata, aggr="sum", debug=debug)
+        self.acc = Accuracy(task="multiclass", num_classes=out_channels)
+
+    def forward(
+        self,
+        x_dict: Dict[NodeType, Tensor],
+        edge_index_dict: Dict[EdgeType, Tensor],
+    ) -> Dict[NodeType, Tensor]:
+        return self.model(x_dict, edge_index_dict)
+
+    def common_step(self, batch: Batch,
+                    ddp_module=None) -> Tuple[Tensor, Tensor]:
```

Comment:
i agree its weird but i needed to do it to make common step work as a function of the model. this is because when you pass the model into DDP it returns the ddp module. then the model, self, has no access to it for the common step. the other way to handle this would be to remove common_step as a method of the model and just pass the module(DDP or regular) every time. i thought this would be even more confusing. this version works, but i am open to suggestions on how to clean this up